### DISCUSSION TITLE
CanalSyncJob什么时候能支持配置zookeeper动态获取canal server的地址以便实现高可用

### DISCUSSION THREAD

**Discussion by xiongzegang** [apache/doris#13331] [A - General / Q&A]
看CanalSyncJob源码里只实现了CanalConnector connector = CanalConnectors.newSingleConnector()，什么时候能把CanalConnector connector = CanalConnectors.newClusterConnector()加上以实现高可用。

**Comment by hf200012** [comment]
这个不是一个好的使用方式，而且只能单独使用MySQL数据，在大数据量负载上也不如直接使用Flink CDC
这个对FE也会产生一些压力，相对这个使用方式我更推荐你使用Flink CDC + Doris Flink Connector，现在Flink的方式我们实现了Exactly-once语义，吞吐借助Flink的能力也能非常好的扩充
你有什么好的想法也可以提出来我们一起改进

**Comment by yegaofei** [comment]
我最近也遇到了这个问题，如果Canal通过ZK做了canal server的多节点高可用，我们是不是可以在Doris里面创建多个CanalSyncJob，分别指向这些cancal server，因为同一时间只会有一个canal server在工作，当一个canal server宕机了，那么另外一个canal server就会开始同步数据，这个时候，对应的CanalSyncJob也会进行切换。 

只是一个想法，没有在生产环境中验证过， 不知道是否可行  @hf200012

**Reply by xiongzegang** [reply]
我有一个pullrequest用来实现高可用，提了很久了，没有合并

**Reply by xiongzegang** [reply]
> 我有一个pullrequest用来实现高可用，提了很久了，没有合并

#13384
